frame/include/bli_type_defs.h

```diff
@@ -1232,6 +1232,47 @@ typedef struct constdata_s
 // -- BLIS object type definitions ---------------------------------------------
 //
 
+// Forward declarations for function pointer types
+struct obj_s;
+struct cntx_s;
+struct rntm_s;
+struct thrinfo_s;
```

Comment:
I didn't notice these during our Zoom review. Is this part of resolving the circular dependency?

Reply:
Yes.

Reply:
Are they needed? What would happen if you removed them?

I feel like in past situations like this, I was able to get by without them, though I defer to you as to explaining why that might have been.

Reply:
`struct obj_s` is definitely needed. The others are also needed unless you want to change the order in which the headers are included in blis.h.

Reply:
Actually, in C maybe you can omit them? I am used to C++ where there is an actual type system :).

Reply:
When you remove them you get:

```
include/haswell/blis.h:1880:14: warning: declaration of 'struct obj_s' will not be visible outside of this function [-Wvisibility]
      struct obj_s* a,
             ^
```

which isn't technically an error but doesn't sound good.

Reply:
Agreed, we should leave them in. I was just curious.